frontend/src/metabase/static-viz/components/LineAreaBarChart/utils/series.tsx

```diff
@@ -168,14 +200,10 @@ function hasTwoDimensions(
   return "breakoutValue" in series;
 }
 
-export function removeNoneSeriesFields(
-  series: (SeriesWithOneOrLessDimensions | SeriesWithTwoDimensions)[][],
-): Series[] {
-  return series
-    .flat()
-    .map(
-      series => _.omit(series, "cardName", "column", "breakoutValue") as Series,
-    );
+export function removeNoneSeriesFields(series: CardSeries): Series[] {
+  return series.map(
+    series => _.omit(series, "cardName", "column", "breakoutValue") as Series,
```

Comment:
move `flat` outside, so I could compose the part that uses this function easier.